pkg/cluster/spec.go

```diff
@@ -168,6 +168,17 @@ func newWithCliVersion(cliVersion version.Info, opts ...SpecOpt) *Spec {
 	return NewSpec(opts...)
 }
 
+func NewSpecFromClusterConfigWithValidate(clusterConfigPath string, cliVersion version.Info, opts ...SpecOpt) (*Spec, error) {
```

Comment:
Curious, do we need to implement this in the `cluster` package? This package already provides the building blocks to implement it and it seems like all the valid usecases come from the cmd package, so I wonder if we should just put it there. My main concern is we are trying to separate these 2 operations, reading and validating. By offering this in the package interface we are kind of perpetuating that pattern. 

I see that test/framework and internal/test both were calling this method, but I don't think either of them need the validation part.

wdyt?

Reply:
Think you're right and going with the cmd pkg will be a good idea to keep the validate and spec creation separate. Adding the function together also resulted in some redundant testing being recommended.

Will move packages.